data/rao-result/rao-result-api/src/main/java/com/powsybl/openrao/data/raoresultapi/RaoResult.java

```diff
@@ -492,28 +494,61 @@ private static byte[] getBytesFromInputStream(InputStream inputStream) throws IO
+     *
+     * @param exporters           candidate CRAC exporters
+     * @param format              desired output CRAC data type
+     * @param cracCreationContext CRAC creation context that contains the original CRAC
+     * @param properties          specific information needed for export
+     * @param outputStream        file where to write the CRAC data
+     */
+    private void write(List<Exporter> exporters, String format, CracCreationContext cracCreationContext, Properties properties, OutputStream outputStream) {
+        exporters.stream()
+            .filter(ex -> format.equals(ex.getFormat()))
+            .findAny()
+            .orElseThrow(() -> new OpenRaoException("Export format " + format + " not supported"))
+            .exportData(this, cracCreationContext, properties, outputStream);
+    }
+
+    /**
+     * Write CRAC data into a file
+     *
+     * @param format              desired output CRAC data type
+     * @param cracCreationContext CRAC creation context that contains the original CRAC
+     * @param properties          specific information needed for export
+     * @param outputStream        file where to write the CRAC data
+     */
+    default void write(String format, CracCreationContext cracCreationContext, Properties properties, OutputStream outputStream) {
+        write(new ServiceLoaderCache<>(Exporter.class).getServices(), format, cracCreationContext, properties, outputStream);
+    }
+
     /**
      * Write CRAC data into a file
      *
      * @param exporters    candidate CRAC exporters
      * @param format       desired output CRAC data type
+     * @param crac         CRAC data
+     * @param properties   specific information needed for export
      * @param outputStream file where to write the CRAC data
      */
-    private void write(List<Exporter> exporters, String format, Crac crac, Set<Unit> flowUnits, OutputStream outputStream) {
+    private void write(List<Exporter> exporters, String format, Crac crac, Properties properties, OutputStream outputStream) {
         exporters.stream()
             .filter(ex -> format.equals(ex.getFormat()))
             .findAny()
             .orElseThrow(() -> new OpenRaoException("Export format " + format + " not supported"))
-            .exportData(this, crac, flowUnits, outputStream);
+            .exportData(this, crac, properties, outputStream);
     }
 
     /**
      * Write CRAC data into a file
      *
      * @param format       desired output CRAC data type
+     * @param crac         CRAC data
+     * @param properties   specific information needed for export
      * @param outputStream file where to write the CRAC data
      */
-    default void write(String format, Crac crac, Set<Unit> flowUnits, OutputStream outputStream) {
-        write(new ServiceLoaderCache<>(Exporter.class).getServices(), format, crac, flowUnits, outputStream);
+    default void write(String format, Crac crac, Properties properties, OutputStream outputStream) {
```

Comment:
I think we shouldn't allow the usage of "raw" crac anymore. @phiedw  ?

Reply:
I agree, I left it there as a temporary workaround because the JSON-CRAC Creation Context is not mature yet

Reply:
ok, leave it for now

Reply:
I gave this question some thinking and what if the CRAC was manually created by the user using adders? There would be no creation context here since the CRAC is not imported
I guess we must leave both signatures anyway